client/src/utils/Server.ts

```diff
@@ -155,6 +155,12 @@ const Server = {
     toggleWar: (id: string) => Fetch<never>(`/api/clan/${id}/toggle-war`, {}, 'PUT'),
     getForAdmin: (id: string) => Fetch<ClanGetForAdminResponse>(`/api/clan/${id}/admin`),
     adminUpdate: (id: string, data: Clan) => Fetch<never>(`/api/clan/${id}/update`, data, 'PATCH'),
+    getRoles: (id: string) => Fetch(`/api/clan/${id}/roles`),
+    createRole: (id: string, data: { name: string;[key: string]: unknown }) => Fetch(`/api/clan/${id}/roles`, data, 'POST'),
+    updateRole: (id: string, roleId: string, data: { name?: string;[key: string]: unknown }) => Fetch(`/api/clan/${id}/roles/${roleId}`, data, 'PUT'),
+    deleteRole: (id: string, roleId: string) => Fetch(`/api/clan/${id}/roles/${roleId}`, {}, 'DELETE'),
+    assignRole: (id: string, bruteId: string, roleId: string) => Fetch(`/api/clan/${id}/members/${bruteId}/roles/${roleId}`, {}, 'POST'),
+    removeRole: (id: string, bruteId: string, roleId: string) => Fetch(`/api/clan/${id}/members/${bruteId}/roles/${roleId}`, {}, 'DELETE'),
```

Comment:
That's where you should set a return type for your fetches, like: `Fetch<MyReturnType>`.
This return type should be declared in the core package, and checked against what your server handlers return to prevent any drift between data sent by the server, and data expected by the client